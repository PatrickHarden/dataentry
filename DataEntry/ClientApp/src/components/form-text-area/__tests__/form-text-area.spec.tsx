import React from 'react';
import { mount, shallow } from 'enzyme';

import FormTextArea from '../form-text-area';

describe('components', () => {

  describe('form-text-area', () => {

    let props:any;
    let wrapper:any;

    beforeEach(() => {
        props = {
            name: 'testme',
            defaultValue: 'text area goodness',
            label: 'I am indeed a text area!',
            error: false
        }
    });

    // it renders it's normal state properly (border should be #cccccc)
    /*
    it("normal state shows normal border (#cccccc)", () => {
        wrapper = mount(<FormTextArea {...props}/>);
        expect(wrapper.find('textarea')).toHaveStyleRule("border","1px solid #cccccc");
    })
    */

    // it renders it's error state properly (border should be red)
    /*
    it("error state shows red border", () => {
        props = {
            name: 'testme',
            defaultValue: 'text area goodness',
            label: 'I am indeed a text area!',
            error: true
        }
        wrapper = mount(<FormTextArea {...props}/>);
        expect(wrapper.find('textarea')).toHaveStyleRule("border","1px solid red");
    })
    */

    // it has a label of "Test Me!"
    it("renders the expected label", () => {
        wrapper = mount(<FormTextArea {...props}/>);
        expect(wrapper.find('h5').text()).toEqual("I am indeed a text area!");
    });

    // it populates it's initial value correctly
    it("populates its initial value", () => {
        wrapper = mount(<FormTextArea {...props}/>);
        expect(wrapper.find('textarea').instance().value).toEqual("text area goodness");
    });

    // snapshot render
    it('renders according to snapshot', () => {
        expect(shallow(<FormTextArea {...props} />)).toMatchSnapshot();
    });
  });
});