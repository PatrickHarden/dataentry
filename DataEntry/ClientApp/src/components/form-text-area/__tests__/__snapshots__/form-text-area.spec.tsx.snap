// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components form-text-area renders according to snapshot 1`] = `
<styled.div>
  <FormFieldLabel
    error={false}
    label="I am indeed a text area!"
  />
  <styled.textarea
    defaultValue="text area goodness"
    error={false}
    id="testme"
    name="testme"
    onBlur={[Function]}
  />
</styled.div>
`;
